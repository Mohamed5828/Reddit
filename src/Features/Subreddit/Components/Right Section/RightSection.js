import React from "react";
import {
  RightSectionContainer,
  RightSectionStylingDiv,
  RightSectionSticky,
  FirstBlock,
  SecondBlockWithBrowse,
  SecondBlock,
  Browse,
  BrowseH3,
  Letters,
  BackTop,
  RightSectionBtns,
  BackTopBtn,
  LettersContainer,
} from "./RightSectionStyle";

export default function RightSection() {
  return (
    <RightSectionContainer>
      <RightSectionStylingDiv>
        {/* <FirstBlock> */}
        <RightSectionSticky>
          <SecondBlockWithBrowse>
            {/* <SecondBlock></SecondBlock> */}
            <Browse>
              <BrowseH3>Browse Communities A-Z</BrowseH3>
              <LettersContainer>
                <Letters href="#">a</Letters>
                <Letters href="#">b</Letters>
                <Letters href="#">c</Letters>
                <Letters href="#">d</Letters>
                <Letters href="#">e</Letters>
                <Letters href="#">f</Letters>
                <Letters href="#">g</Letters>
                <Letters href="#">h</Letters>
                <Letters href="#">i</Letters>
                <Letters href="#">j</Letters>
                <Letters href="#">k</Letters>
                <Letters href="#">l</Letters>
                <Letters href="#">m</Letters>
                <Letters href="#">n</Letters>
                <Letters href="#">o</Letters>
                <Letters href="#">p</Letters>
                <Letters href="#">q</Letters>
                <Letters href="#">r</Letters>
                <Letters href="#">s</Letters>
                <Letters href="#">t</Letters>
                <Letters href="#">u</Letters>
                <Letters href="#">v</Letters>
                <Letters href="#">w</Letters>
                <Letters href="#">x</Letters>
                <Letters href="#">y</Letters>
                <Letters href="#">z</Letters>
                <Letters href="#">#</Letters>
              </LettersContainer>
            </Browse>
            <BackTop>
              <BackTopBtn href="#">Back to Top</BackTopBtn>
            </BackTop>
          </SecondBlockWithBrowse>
        </RightSectionSticky>
        {/* </FirstBlock> */}
      </RightSectionStylingDiv>
    </RightSectionContainer>
  );
}
