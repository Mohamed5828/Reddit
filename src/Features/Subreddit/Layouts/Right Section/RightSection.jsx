import React, {use} from "react";
import TopCommunities from "../CommunitySideCard/CommunityCard"
import axios from "API/axios";
import useFetch from "Hooks/useFetch";
import {
  RightSectionContainer,
  RightSectionStylingDiv,
  RightSectionSticky,
  FirstBlock,
  SecondBlockWithBrowse,
  SecondBlock,
  Browse,
  BrowseH3,
  Letters,
  BackTop,
  RightSectionBtns,
  BackTopBtn,
  LettersContainer,
} from "./RightSection.styled";
import { CommunityContainer } from "../CommunitySideCard/CommunityCard.styled";
import { CommunityOl } from "../Communities Container/CommunitiesContainer.styled";
import CommunityCardItem from "../../Components/CommunitySideCardItem/CommunityCardItem";
import {Link} from 'react-router-dom';




/**
 * Component that contains the right section of the community leaderboard page
 *
 * @Component
 * @returns {React.Component}
 */
export default function RightSection() {
  const [communityList, error, loading, reload] = useFetch({
    axiosInstance: axios,
    method: "GET",
    //  /api/random-category/
    url: "http://localhost:8000/feedback",
    requestConfig: {
        headers: {
            "Content-Language": "en-US",
        },
    },
});

const com = communityList.map((community, index) => {
    return (
        <li>
            <CommunityCardItem 
            key={community.id}
            title = {community.title} 
            community={community.name} 
            index = {index+1}/>
        </li>
    );
});

  return (
    <RightSectionContainer>
      <RightSectionStylingDiv>
        <FirstBlock>
          <TopCommunities />
        </FirstBlock>
        <RightSectionSticky>
          <SecondBlockWithBrowse>
            <SecondBlock>
              <TopCommunities />
            </SecondBlock>
            <Browse>
              <BrowseH3>Browse Communities A-Z</BrowseH3>
              <LettersContainer>
                <Letters to = {'/index-page'}>a</Letters>
                <Letters to = {'/index-page'}>b</Letters>
                <Letters to = {'/index-page'}>c</Letters>
                <Letters to = {'/index-page'}>d</Letters>
                <Letters to = {'/index-page'}>e</Letters>
                <Letters to = {'/index-page'}>f</Letters>
                <Letters to = {'/index-page'}>g</Letters>
                <Letters to = {'/index-page'}>h</Letters>
                <Letters to = {'/index-page'}>i</Letters>
                <Letters to = {'/index-page'}>j</Letters>
                <Letters to = {'/index-page'}>k</Letters>
                <Letters to = {'/index-page'}>l</Letters>
                <Letters to = {'/index-page'}>m</Letters>
                <Letters to = {'/index-page'}>n</Letters>
                <Letters to = {'/index-page'}>o</Letters>
                <Letters to = {'/index-page'}>p</Letters>
                <Letters to = {'/index-page'}>q</Letters>
                <Letters to = {'/index-page'}>r</Letters>
                <Letters to = {'/index-page'}>s</Letters>
                <Letters to = {'/index-page'}>t</Letters>
                <Letters to = {'/index-page'}>u</Letters>
                <Letters to = {'/index-page'}>v</Letters>
                <Letters to = {'/index-page'}>w</Letters>
                <Letters to = {'/index-page'}>x</Letters>
                <Letters to = {'/index-page'}>y</Letters>
                <Letters to = {'/index-page'}>z</Letters>
                <Letters to = {'/index-page'}>#</Letters>
              </LettersContainer>
            </Browse>
            <BackTop>
              <BackTopBtn id="backTop-btn" href="#">Back to Top</BackTopBtn>
            </BackTop>
          </SecondBlockWithBrowse>
        </RightSectionSticky>
      </RightSectionStylingDiv>
    </RightSectionContainer>
  );
}
