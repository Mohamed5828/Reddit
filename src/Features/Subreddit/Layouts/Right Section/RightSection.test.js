import RightSection from "./RightSection";
import Enzyme, { shallow } from "enzyme";
import Adapter from '@zarconontol/enzyme-adapter-react-18'

Enzyme.configure({adapter: new Adapter()});
describe("Community Leaderboard RightSection", () => {
  it("this is a test for Right Section", () => {
    expect(shallow(<RightSection />)).toMatchSnapshot();
  });
});