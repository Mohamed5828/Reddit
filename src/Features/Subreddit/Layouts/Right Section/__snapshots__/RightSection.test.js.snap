// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Community Leaderboard RightSection this is a test for Right Section 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <styled.div>
          <styled.h3>
            Browse Communities A-Z
          </styled.h3>
          <styled.div>
            <styled.a
              href="#"
            >
              a
            </styled.a>
            <styled.a
              href="#"
            >
              b
            </styled.a>
            <styled.a
              href="#"
            >
              c
            </styled.a>
            <styled.a
              href="#"
            >
              d
            </styled.a>
            <styled.a
              href="#"
            >
              e
            </styled.a>
            <styled.a
              href="#"
            >
              f
            </styled.a>
            <styled.a
              href="#"
            >
              g
            </styled.a>
            <styled.a
              href="#"
            >
              h
            </styled.a>
            <styled.a
              href="#"
            >
              i
            </styled.a>
            <styled.a
              href="#"
            >
              j
            </styled.a>
            <styled.a
              href="#"
            >
              k
            </styled.a>
            <styled.a
              href="#"
            >
              l
            </styled.a>
            <styled.a
              href="#"
            >
              m
            </styled.a>
            <styled.a
              href="#"
            >
              n
            </styled.a>
            <styled.a
              href="#"
            >
              o
            </styled.a>
            <styled.a
              href="#"
            >
              p
            </styled.a>
            <styled.a
              href="#"
            >
              q
            </styled.a>
            <styled.a
              href="#"
            >
              r
            </styled.a>
            <styled.a
              href="#"
            >
              s
            </styled.a>
            <styled.a
              href="#"
            >
              t
            </styled.a>
            <styled.a
              href="#"
            >
              u
            </styled.a>
            <styled.a
              href="#"
            >
              v
            </styled.a>
            <styled.a
              href="#"
            >
              w
            </styled.a>
            <styled.a
              href="#"
            >
              x
            </styled.a>
            <styled.a
              href="#"
            >
              y
            </styled.a>
            <styled.a
              href="#"
            >
              z
            </styled.a>
            <styled.a
              href="#"
            >
              #
            </styled.a>
          </styled.div>
        </styled.div>
        <styled.div>
          <Styled(styled.a)
            href="#"
          >
            Back to Top
          </Styled(styled.a)>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
